### ISSUE TITLE
mozjpeg failing on OS X / macOS and possibly others

### ISSUE THREAD

**Issue by samit4me** [mxstbr/react-boilerplate#1335]
`mozjpeg` is a dependency of `image-webpack-loader@3.0.0`!

The problem is that `mozjpeg` appears to have some environmental dependencies that doen't exist by default on OS X / macOS, see [issue here](https://github.com/tcoopman/image-webpack-loader/issues/49).

The "solution" is to use `image-webpack-loader@2.0.0` as it uses `jpegtran` instead of `mozjpeg`.  `image-webpack-loader` made the switch in v3 as you can see in [this commit](https://github.com/tcoopman/image-webpack-loader/commit/0cad7983fdae77f5f532a2725a2d997e24949824#diff-b9cfc7f2cdf78a7f4b91a753d10865a2).

**Comment by andykenward** [comment]
I found that i had to install `libpng` to solve this issue on macOS `10.12.1`

If you have brew
`brew install libpng`

**Comment by gihrig** [comment]
As covered in [#1322 comment](https://github.com/mxstbr/react-boilerplate/pull/1322#issuecomment-266267339) this error is found on some OS X installations, it does not seem to be version specific.

As an interim solution I have added to the dependency update notes that RBP will remain at `image-webpack-loader@2.0.0` until the upstream issue is resolved.

**Comment by samit4me** [comment]
Thanks @andykenward , I will try that tonight. Even if this does work, I'm not sure requiring people install stuff outside of (Yarn / npm) is a great solution!

**Comment by samit4me** [comment]
I can confirm that `brew install libpng` has resolved the problem, thanks @andykenward!

Now to decide if that is an acceptable solution OR do we use v2 until something upstream changes?

**Comment by gihrig** [comment]
> I'm not sure requiring people install stuff outside of (Yarn / npm) is a great solution!

I totally agree. To say (document) in `gotchas.md` Mac OS: This project _may_ require you to install libpng:

```bach
 brew install libpng
```

Seems rather specialized considering that RBP is not graphics focused.

I vote for holding image-webpack-loader@2.0.0 until the upstream issue is resolved.

**Comment by samit4me** [comment]
@gihrig You seem to handle all the dependencies around here, how would you like to handle this?

I assume we go back to v2 and then check on the upstream progress each time the deps are updated?

**Comment by mxstbr** [comment]
> I assume we go back to v2 and then check on the upstream progress each time the deps are updated?

Yes, that sounds like the way it'll have to go! Thanks for digging in everybody, would appreciate a PR pinning this to v2!

**Comment by gihrig** [comment]
Agreed, I'll keep an eye on the upstream [issue](https://github.com/tcoopman/image-webpack-loader/issues/49) and hold at image-webpack-loader@2.0.0 until a resolution appears.

**Comment by gihrig** [comment]
The core issue seems to be Addressed in [imagemin/imagemin-pngquant#32](https://github.com/imagemin/imagemin-pngquant/issues/32)

But... 

Warning: Observed on npm start:
```bash
WARNING in ./app/components/Header/banner.jpg
DEPRECATED. Configure gifsicle's interlaced option in it's own options. (gifsicle.interlaced)
@ ./app/components/Header/index.js 47:0-34
@ ./app/containers/App/index.js
@ ./app/app.js
@ multi eventsource-polyfill webpack-hot-middleware/client?reload=true ./app/app.js
WARNING in ./app/components/Header/banner.jpg
DEPRECATED. Configure optipng's optimizationLevel option in it's own options. (optipng.optimizationLevel)
@ ./app/components/Header/index.js 47:0-34
@ ./app/containers/App/index.js
@ ./app/app.js
@ multi eventsource-polyfill webpack-hot-middleware/client?reload=true ./app/app.js

image-webpack-loader 2.0.0 → 3.2.0 <--- Rolled back
```

Because `gifsicle` and `optipng` are not explicit dependencies of RBP... Holding at `image-webpack-loader 2.0.0` for now.
